usdigitalresponse/usdr-gost
# Cypress: Adds ability to run and open integration test suite locally

### Ticket #1066 
## Description
This PR does a few things:
1. Installs Cypress package
2. Allows user to `yarn run cypress:run` and `yarn run cypress:open`
3. Scaffolding to write new cypress specs
4. Integrate Cypress tests into a github workflow when PRs are opened.


What the PR does not do:
1. Documentation on how to use Cypress (This will come in a separate PR)

## Screenshots / Demo Video
N/A - I'll be posting a loom video as part of documentation in future.

## Testing
To test manually
1. Pull down the repository
2. Ensure database is migrated and front-end is serving on `localhost:8080` and `locahost:3000`
3. Run `yarn cypress:run` and see that tests are passing
4. Go through this setup process: https://sourabhbajaj.com/blog/2017/02/07/gui-applications-docker-mac/?ref=cypress-blog
5. Run `yarn cypress:up` to see that the interactive Cypress is running
6. Run `yarn cypress:down` to ensure that the containers are cleaned up.

### Automated and Unit Tests
N/A

### Manual tests for Reviewer

## Checklist
